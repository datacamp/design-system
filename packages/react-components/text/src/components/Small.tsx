import tokens from '@datacamp/waffles-tokens';
import { computeDataAttributes } from '@datacamp/waffles-utils';
import { css } from '@emotion/react';
import React from 'react';

import baseStyle from '../baseStyle';

interface SmallProps {
  /**
   * The content to display. Can contain a combination of strings, Strong
   * components, and Emphasis Components.
   */
  children: React.ReactNode;
  /**
   * Sets the css class of the rendered element. Can be used to apply custom
   * styles.
   */
  className?: string;
  /**
   * As with all the other waffles components, dataAttributes can be used to set
   * data- html attributes on the element.
   */
  dataAttributes?: { [key: string]: string };
}

const smallStyle = css(baseStyle, {
  fontSize: tokens.fontSizes.small,
  lineHeight: tokens.lineHeights.relaxed,
});

const Small = ({
  children,
  className,
  dataAttributes,
}: SmallProps): JSX.Element => {
  const parsedDataAttributes = computeDataAttributes(dataAttributes);

  return (
    <small className={className} css={smallStyle} {...parsedDataAttributes}>
      {children}
    </small>
  );
};

export default Small;
