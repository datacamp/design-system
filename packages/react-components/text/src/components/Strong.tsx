import tokens from '@datacamp/waffles-tokens/lib/future-tokens.json';
import { css } from '@emotion/core';
import React from 'react';

import baseStyle from '../baseStyle';
import computeDataAttributes from '../computeDataAttributes';

interface StrongProps {
  children: string;
  className?: string;
  dataAttributes?: { [key: string]: string };
}

const strongStyle = css(baseStyle, {
  fontWeight: tokens.size.fontWeight.bold.value,
  lineHeight: tokens.size.font.base.value,
});

const Strong: React.FC<StrongProps> = ({
  children,
  className,
  dataAttributes,
}) => {
  const parsedDataAttributes = computeDataAttributes(dataAttributes);

  return (
    <strong className={className} css={strongStyle} {...parsedDataAttributes}>
      {children}
    </strong>
  );
};

export default Strong;
