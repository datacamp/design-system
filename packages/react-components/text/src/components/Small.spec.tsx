import 'jest-dom/extend-expect';

import { render } from '@testing-library/react';
import React from 'react';

import Small from './Small';
import Strong from './Strong';

const testText = 'Some example text.';
const testClassName = 'example-class';
const testDataAttributes = { cy: 'othertest', example: 'testdata' };

describe('<Small />', () => {
  it('renders with a string', () => {
    const { container } = render(
      <Small className={testClassName} dataAttributes={testDataAttributes}>
        {testText}
      </Small>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders with Strong children', () => {
    const { container } = render(
      <Small className={testClassName} dataAttributes={testDataAttributes}>
        Some example text <Strong>with strong</Strong> included.
      </Small>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('sets the text in the span', () => {
    const { getByText } = render(<Small>{testText}</Small>);
    expect(getByText(testText)).toBeDefined();
  });

  it('sets the className', () => {
    const { getByText } = render(
      <Small className={testClassName}>{testText}</Small>
    );
    expect(getByText(testText)).toHaveClass(testClassName);
  });

  it('sets the data attributes', () => {
    const { getByText } = render(
      <Small dataAttributes={testDataAttributes}>{testText}</Small>
    );
    expect(getByText(testText)).toHaveAttribute(
      'data-cy',
      testDataAttributes.cy
    );
    expect(getByText(testText)).toHaveAttribute(
      'data-example',
      testDataAttributes.example
    );
  });
});
