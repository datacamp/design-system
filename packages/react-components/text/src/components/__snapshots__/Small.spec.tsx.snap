// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Small /> renders with Strong and Emphasis children 1`] = `
.emotion-2 {
  -webkit-font-smoothing: antialiased;
  color: rgb(5,25,45);
  font-family: Arial;
  font-family: Studio-Feixen-Sans-Variable;
  font-style: normal;
  font-size: 14px;
  line-height: 16px;
}

.emotion-0 {
  -webkit-font-smoothing: antialiased;
  color: rgb(5,25,45);
  font-family: Arial;
  font-family: Studio-Feixen-Sans-Variable;
  font-style: normal;
  font-weight: 800;
  line-height: 16px;
}

.emotion-1 {
  -webkit-font-smoothing: antialiased;
  color: rgb(5,25,45);
  font-family: Arial;
  font-family: Studio-Feixen-Sans-Variable;
  font-style: normal;
  font-style: italic;
  font-weight: 400;
  line-height: 16px;
}

<small
  class="example-class emotion-2"
  data-example="testdata"
  data-item="othertest"
>
  Some example text 
  <strong
    class="emotion-0"
  >
    with strong
  </strong>
   and
   
  <em
    class="emotion-1"
  >
    emphasis text
  </em>
   included.
</small>
`;

exports[`<Small /> renders with a string 1`] = `
.emotion-0 {
  -webkit-font-smoothing: antialiased;
  color: rgb(5,25,45);
  font-family: Arial;
  font-family: Studio-Feixen-Sans-Variable;
  font-style: normal;
  font-size: 14px;
  line-height: 16px;
}

<small
  class="example-class emotion-0"
  data-example="testdata"
  data-item="othertest"
>
  Some example text.
</small>
`;
